### ISSUE TITLE
github.com/openshift/origin/test/integration.TestPullThroughInsecure

### ISSUE THREAD

**Issue by soltysh** [openshift/origin#14434]
Seen it in: https://ci.openshift.redhat.com/jenkins/job/test_pull_request_origin_integration/2741/

The root cause seems to be `concurrent map writes`, check [here](https://ci.openshift.redhat.com/jenkins/job/test_pull_request_origin_integration/2741/testReport/junit/github/com_openshift_origin_test_integration%20%20%20/TestPullThroughInsecure/) for full details.

**Comment by soltysh** [comment]
Hmmm, looks like this:

```
goroutine 7306 [running]:
runtime.throw(0x53417fc, 0x15)
	/usr/local/go/src/runtime/panic.go:566 +0x95 fp=0xc43e33c338 sp=0xc43e33c318
runtime.mapassign1(0x4a83400, 0xc43f46c1b0, 0xc43e33c520, 0xc43e33c510)
	/usr/local/go/src/runtime/hashmap.go:458 +0x8ef fp=0xc43e33c420 sp=0xc43e33c338
github.com/openshift/origin/pkg/dockerregistry/server.(*remoteBlobGetterService).proxyStat(0xc43f457200, 0x7f1480c6b680, 0xc43f4c3770, 0x9bbeca0, 0xc43f58e820, 0xc43e33c998, 0xc43f36eba9, 0x47, 0x0, 0x0, ...)
	/openshifttmp/openshift/build-rpm-release/tito/rpmbuild-originbxeiT0/BUILD/origin-3.6.0/_output/local/go/src/github.com/openshift/origin/pkg/dockerregistry/server/remoteblobgetter.go:181 +0xcf9 fp=0xc43e33c8b0 sp=0xc43e33c420
```
might be causing this.

**Comment by miminar** [comment]
On it.

**Comment by miminar** [comment]
Fix: #14476
